java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -24,18 +24,86 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.xml.datatype.DatatypeConfigurationException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 import java.util.function.Consumer;
 
+import static ai.rapids.cudf.HostColumnVector.OFFSET_SIZE;
+
 /**
  * This class represents the immutable vector of data.  This class holds
  * references to device(GPU) memory and is reference counted to know when to release it.  Call
  * close to decrement the reference count when you are done with the column, and call incRefCount
  * to increment the reference count.
  */
-public final class ColumnVector implements AutoCloseable, BinaryOperable {
+public final class ColumnVector implements AutoCloseable, BinaryOperable, ColumnViewPointerAccess {
   private static final Logger log = LoggerFactory.getLogger(ColumnVector.class);
 
+  public class DeviceColumnViewPointerAccess implements ColumnViewPointerAccess {
+
+    protected long viewHandle;
+
+    public DeviceColumnViewPointerAccess(long viewHandle) {
+      this.viewHandle = viewHandle;
+    }
+
+    @Override
+    public long getColumnView() {
+      return viewHandle;
+    }
+
+    @Override
+    public ColumnViewPointerAccess getChildColumnView() {
+      if (getDataType() != DType.LIST) {
+        return null;
+      }
+      long childColumnView = getChildCvPointer(viewHandle);
+      return new DeviceColumnViewPointerAccess(childColumnView);
+    }
+
+    @Override
+    public MemoryBuffer getDataBuffer() {
+      return offHeap.getData();
+    }
+
+    @Override
+    public MemoryBuffer getOffsetBuffer() {
+      return offHeap.getNativeOffsetsPointer(viewHandle);
+    }
+
+    @Override
+    public MemoryBuffer getValidityBuffer() {
+      return offHeap.getNativeValidPointer(viewHandle);
+    }
+
+    @Override
+    public DType getDataType() {
+      return offHeap.getNativeType(viewHandle);
+    }
+
+    @Override
+    public long getRows() {
+      return offHeap.getNativeRowCount(viewHandle);
+    }
+
+    @Override
+    public void close() {
+      if (getDataType() == DType.LIST) {
+        ColumnViewPointerAccess child = getChildColumnView();
+        if (child != null) {
+          child.close();
+        }
+      }
+      deleteColumnView(viewHandle);
+    }
+  }
+
+  private List<MemoryBuffer> getToClose() {
```

Comment:
@revans2 , are we ok with passing the `NestedColumnVector` to the `ColumnVector` constructor so that it can then pick out the additional buffers it needs to track and the viewhandle (which is passed in right now to allow `makeCudfColumnView`) ?